components/tabs/Tabs.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { FC, useState, ReactElement } from "react";
+import Tab from "./Tab";
+import { TabList } from "./Tabs.styled";
+import { TabProps } from "./Tab";
+interface Props {
+  children: Array<ReactElement<TabProps>>;
+}
+
+const Tabs: FC<Props> = ({ children }) => {
+  const [activeTab, setActiveTab] = useState<string>(
+    children[0].props["data-label"]
+  );
+
+  const onClickTabItem = (tab: string) => {
+    setActiveTab(tab);
+  };
+
+  return (
+    <div>
+      <TabList>
+        {children.map((child) => {
+          return (
+            <Tab
+              key={child.props["data-label"]}
+              activeTab={activeTab}
+              data-label={child.props["data-label"]}
+              onClick={onClickTabItem}
+            />
+          );
+        })}
+      </TabList>
+      <div className="tab-content">
+        {children.map((child) => {
+          if (child.props["data-label"] !== activeTab) return undefined;
+          return child;
```

Comment:
this returns an array with a single defined value and a bunch of undefineds, react interprets this correctly, ie it only shows the defined child?

Reply:
Yeah so what's actually happening here is you are iterating over an array of elements.

If you return undefined it won't render (or null) -- basically, you need to return something valid JSX or JSX-like it to actually render. 

In this case, we just want the content that matches the active tab, so if this comparison matches, we return the child.

Reply:
interesting, ok thats cool